src/hotspot/share/interpreter/bytecodeUtils.cpp

```diff
@@ -353,7 +353,7 @@ static void print_local_var(outputStream *os, unsigned int bci, Method* method,
 
 StackSlotAnalysisData::StackSlotAnalysisData(BasicType type) : _bci(INVALID), _type(type) {}
 
-StackSlotAnalysisData::StackSlotAnalysisData(int bci, BasicType type) : _bci(bci), _type(type) {
+StackSlotAnalysisData::StackSlotAnalysisData(int bci, BasicType type) : _bci((u2)bci), _type(type) {
```

Comment:
Looking at this more closely, it seems like bci is meant to be a u2 but it needs one extra bit for the "INVALID" value. Maybe it's possible to represent invalid bci's differently? That way we can just set _bci to be a u2.

Reply:
If I pass u2 bci, it'll fit in the field.  The INVALID bit is added after bci - ie the bci is never INVALID.